domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmAddHandler.java

```diff
@@ -125,7 +128,17 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
         // Install another RUNTIME handler to actually install the services. This will run after the
         // RUNTIME handler for any child resources. Doing this will ensure that child resource handlers don't
         // see the installed services and can just ignore doing any RUNTIME stage work
-        context.addStep(ServiceInstallStepHandler.INSTANCE, OperationContext.Stage.RUNTIME);
+        if(!context.isBooting() && context.getProcessType() == ProcessType.EMBEDDED_SERVER && context.getRunningMode() == RunningMode.ADMIN_ONLY) {
+            context.reloadRequired();
+            context.completeStep(new OperationContext.RollbackHandler() {
```

Comment:
Don't add a rollback handler, override rollbackRuntime, which is called by the RollbackHandler added by the superclass.